topostats/find_grains.py

```diff
@@ -0,0 +1,220 @@
+def get_lower_threshold(image: np.array, lower_threshold_multiplier: float = 1.7) -> float:
+    """Calculate the lower threshold for filtering.
+
+    Parameters
+    ----------
+    image: np.array
+        Numpy array representing image.
+    lower_threshold_multiplier: float
+        Factor by which threshold is to be scaled by.
+    Returns
+    -------
+    float
+        Lower threshold
+    """
+    LOGGER.info('Calculating lower threshold for grain identification.')
+    return get_threshold(image) * lower_threshold_multiplier
+
+
+def gaussian_filter(image: np.array, gaussian_size: float = 2, dx: float = 1, mode: str = 'nearest') -> np.array:
+    """Apply Gaussian filter
+
+    Parameters
+    ----------
+    image : np.array
+        Numpy array of image.
+    gaussian_size : float
+        Gaussian blur size in nanometers (nm).
+    dx : float
+        Pixel to nanometer scale.
+    mode : str
+        Mode for filtering (default is 'nearest')
+
+    Returns
+    -------
+    np.array
+        Numpy array of filtered image.
+
+    Examples
+    --------
+    FIXME: Add docs.
+    """
+    LOGGER.info('Applying Gaussian filter (mode : {mode}; Gaussian blur (nm) : {gaussian_size}).')
+    return gaussian(image, sigma=(gaussian_size / dx), mode=mode)
+
+
+def boolean_image(image: np.array, threshold: float) -> np.array:
+    """Create a boolean array of whether points are greater than the given threshold.
+
+    Parameters
+    ----------
+    image: np.array
+        Numpy array representing image.
+    threshold: float
+        Threshold for masking points in the image.
+    Returns
+    -------
+    np.array
+        Numpy array for masking
+    """
+    LOGGER.info('Created boolean image')
+    return np.array(np.copy(image) > threshold, dtype='bool')
+
+
+def tidy_border(image: np.array) -> np.array:
+    """Remove grains touching the border
+    Parameters
+    ----------
+    image: np.array
+        Numpy array representing image.
+
+    Returns
+    -------
+    np.array
+        Numpy array of image with borders tidied.
+    """
+    LOGGER.info('Tidying borders')
+    return clear_border(np.copy(image))
+
+
+# def calc_minimum_grain_size(minimum_grain_size: float = None, dx: float = 1) -> float:
+#     """Calculate minimum grain size in pixels.
+#
+#     """
+
+
+def remove_objects(image: np.array, minimum_grain_size: float, dx: float) -> np.array:
+    """Remove small objects
+
+    Parameters
+    ----------
+    image: np.array
+        Numpy array representing image.
+    minimum_grain_size: float
+        Minimum grain size in nanometers.
+    dx: float
+        Pixel to nanometer scale.
+    Returns
+    -------
+    np.array
+        Numpy array of image with objects coloured.
+    """
+    LOGGER.info(f'Removing small objects (< {minimum_grain_size / dx})')
+    return remove_small_objects(image, min_size=(minimum_grain_size / dx))
+
+
+def label_regions(image: np.array, background: float = 0.0) -> np.array:
```

Comment:
Does this add much to the function it's wrapping? Do we need it?

Reply:
For this and other functions I was intending to add `**kwargs` to these as I move from functions to classes and methods, allowing users to modify the calls to `label()`, `lab2rgb()`, `regionprops()`, which is why I wrap them up in a function.

At the moment I don't see much scope for people tweaking things to this level, but figured its useful to build this level of flexibility in from the outset.